apps/spruce/src/components/Settings/EventLog/EventDiffTable/utils.test.ts

```diff
@@ -0,0 +1,238 @@
+    expect(getChangedPaths(result)).toEqual(["foo", "nested.key"]);
+  });
+
+  it("returns deleted fields as changed paths", () => {
+    const oldObj = {
+      foo: "bar",
+      nested: { key: 42 },
+    };
+    const newObj = {};
+
+    const result = diff(oldObj, newObj);
+    // deep-object-diff will return the minimal set of changes
+    // so we only get the top-level keys that were deleted
+    expect(getChangedPaths(result)).toEqual(["foo", "nested"]);
+  });
+
+  it("returns empty array if no changes", () => {
+    const obj = {
+      a: 1,
+      b: {
+        c: 2,
+      },
+    };
+
+    const result = diff(obj, { ...obj });
+    expect(getChangedPaths(result)).toEqual([]);
+  });
+});
+
+const exampleAddition = {
+  before: {
+    __typename: "ProjectEventSettings",
+    projectRef: {
+      __typename: "Project",
+      identifier: "viewTest",
+      patchTriggerAliases: [],
+    },
+    vars: {
+      __typename: "ProjectVars",
+      vars: { newVariable: "so new" },
+    },
+  },
+  after: {
+    __typename: "ProjectEventSettings",
+    projectRef: {
+      __typename: "Project",
+      identifier: "viewTest",
+      patchTriggerAliases: [
+        {
+          __typename: "PatchTriggerAlias",
+          alias: "newAlias",
+          childProjectIdentifier: "evg",
+        },
+      ],
+    },
+    vars: {
+      __typename: "ProjectVars",
+      vars: { newVariable: "so new" },
+    },
+  },
+};
+
+const exampleUpdate = {
+  before: {
+    __typename: "ProjectEventSettings",
+    projectRef: {
+      __typename: "Project",
+      identifier: "viewTest",
+      patchTriggerAliases: [
+        {
+          __typename: "PatchTriggerAlias",
+          alias: "newAlias",
+          childProjectIdentifier: "evg",
+        },
+      ],
+    },
+    vars: {
+      __typename: "ProjectVars",
+      vars: { newVariable: "so new" },
+    },
+  },
+  after: {
+    __typename: "ProjectEventSettings",
+    projectRef: {
+      __typename: "Project",
+      identifier: "viewTest",
+      patchTriggerAliases: [
+        {
+          __typename: "PatchTriggerAlias",
+          alias: "noLongerNewAlias",
+          childProjectIdentifier: "evg",
+        },
+      ],
+    },
+    vars: {
+      __typename: "ProjectVars",
+      vars: { newVariable: "so new" },
+    },
+  },
+};
+
+const exampleDeletion = {
+  before: {
+    __typename: "ProjectEventSettings",
+    projectRef: {
+      __typename: "Project",
+      identifier: "viewTest",
+      patchTriggerAliases: [],
+    },
+    vars: {
+      __typename: "ProjectVars",
+      vars: { newVariable: "so new" },
+    },
+  },
+  after: {
+    __typename: "ProjectEventSettings",
+    projectRef: {
+      __typename: "Project",
+      identifier: "viewTest",
+      patchTriggerAliases: [],
+    },
+    vars: {
+      __typename: "ProjectVars",
+      vars: {},
+    },
+  },
+};
+
+describe("should transform event diffs to key, before and after", () => {
```

Comment:
nit. can we name the test after the function it is testing
```suggestion
describe("getEventDiffLines", () => {
```